dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -3784,6 +3784,9 @@ bool MergeTreeData::selectPartsAndMove()
 {
     if (parts_mover.moves_blocker.isCancelled())
         return false;
+    std::unique_lock<std::mutex> lock(storage_policy->moving_parts_mutex, std::try_to_lock);
```

Comment:
This code is executed in `BackgroundProcessingPool` which run tasks concurrently. Parts move is a quite heavy operation and we cannot execute them sequentially. Also why we try to lock, instead of just take it or wait?

Reply:
`just take it or wait` will result in all background thread waiting on the lock.
Maybe a fine-grained lock is better,  change the lock from StoragePolicy to IDisk?